packages/react-reconciler/src/ReactFiberClassComponent.new.js

```diff
@@ -888,9 +889,12 @@ function mountClassInstance(
     processUpdateQueue(workInProgress, newProps, instance, renderLanes);
     instance.state = workInProgress.memoizedState;
   }
-
   if (typeof instance.componentDidMount === 'function') {
-    workInProgress.flags |= Update;
+    if (enableDoubleInvokingEffects) {
```

Comment:
Poo. I realize this was the diff I shared via the other comment, but now that I see it– should we make these conditionals `__DEV__ && enableDoubleInvokingEffects` to match the other place this flag is used?